core/src/test/java/org/commonjava/maven/ext/manip/util/PropertiesUtilsTest.java

```diff
@@ -15,33 +15,219 @@
     }
 
     @Test
     public void testCacheProperty() throws Exception
     {
         Map propertyMap = new HashMap();
 
-        assertFalse(PropertiesUtils.cacheProperty( propertyMap, null, "2.0", null, false ));
-        assertFalse(PropertiesUtils.cacheProperty( propertyMap, "1.0", "2.0", null, false ));
-        assertTrue(PropertiesUtils.cacheProperty( propertyMap, "${version.org.jboss}", "2.0", null, false ));
+        assertFalse( PropertiesUtils.cacheProperty( null, "${foobar}${foobar2}", null, null, false ) );
+        assertFalse( PropertiesUtils.cacheProperty( null, "suffix.${foobar}", null, null, false ) );
+        assertFalse( PropertiesUtils.cacheProperty( propertyMap, null, "2.0", null, false ) );
+        assertFalse( PropertiesUtils.cacheProperty( propertyMap, "1.0", "2.0", null, false ) );
+        assertTrue( PropertiesUtils.cacheProperty( propertyMap, "${version.org.jboss}", "2.0", null, false ) );
+        assertFalse ( PropertiesUtils.cacheProperty( propertyMap, "${project.version}", "2.0", null, false ) );
+
+        // DependencyManipulator does dependency.getVersion(). This could return e.g. ${version.scala} which can
+        // refer to <version.scala>${version.scala.major}.7</version.scala>. If we are attempting to change version.scala
+        // to e.g. 2.11.7.redhat-1 then in this case we need to ignore the version.scala.major property and append the .redhat-1.
+
+        // If the property is ${...}.foobar then we only want to append suffix to foobar to change the version
+        // However we don't need to change the value of the property. If the property is foobar.${....} then
+        // we want to append suffix to the property ... but we need to handle that part of the property is hardcoded.
+
+        assertFalse( PropertiesUtils.cacheProperty( propertyMap, "${version.scala}.7", "2.0", null, false ) );
+        assertFalse( PropertiesUtils.cacheProperty( propertyMap, "${version.foo}.${version.scala}.7", "2.0", null, false ) );
+
+        try
+        {
+            PropertiesUtils.cacheProperty( propertyMap, "${version.scala}.7.${version.scala2}", "2.0", null, false );
+        }
+        catch (ManipulationException e)
+        {
+            // Pass.
+        }
     }
 
+    private ManipulationSession createUpdateSession()
+    {
+        ManipulationSession session = new ManipulationSession();
+        Properties p = new Properties();
+        p.setProperty( "strictAlignment", "true" );
+        p.setProperty( "strictViolationFails", "true" );
+        p.setProperty( "version.suffix", "redhat-1" );
+        session.setState( new DependencyState( p ) );
+        session.setState( new VersioningState( p ) );
+
+        return session;
+    }
+
+    @Test
+    public void testUpdateNestedProperties() throws Exception
+    {
+        final Model modelParent = resolveRemoteModel( "org.infinispan:infinispan-bom:8.2.0.Final" );
+        Project pP = new Project( modelParent );
```

Comment:
Is there any chance of just adding these to src/test/resources to avoid test failures due to network conditions? That makes it easier to work on PME while you're on a train, too... ;-D